### DISCUSSION TITLE
Different behavior than standard Platform IO 'pico' when running simple Modbus RTU Server example using ArduinoModbus

### DISCUSSION THREAD

**Discussion by pomplesiegel** [earlephilhower/arduino-pico#2379] [General]
Hello! I realize this is crossing streams slightly, but I'm wondering if what's going on here may be on a lower embedded level. 

I am building a device using an Adafruit Feather RP2040 + a RS485 interface for Modbus RTU communication. Everything was working well when using the standard [Raspberry Pi Pico](https://docs.platformio.org/en/latest/platforms/raspberrypi.html) platform on Platform IO. However, I really prefer the arduino-pico platform so I changed over. Everything was working great, but for some strange reason only 1 modbus server/slave running `arduino-pico` can be reached on a RS485 bus without waiting multiple seconds - a totally new and very strange issue.

I'm wondering if some strange timing / pin behavior is happening here under the hood? 

This is the [issue](https://github.com/arduino-libraries/ArduinoModbus/issues/155) I created for the ArduinoModbus library, including a very short 20-ish-line generic MVP example of the issue. 

Anyone have any ideas on what could be going on here? I've been beating my head against a wall trying experiments to uncover what could be the underlying issue. 

Thank you!
Michael

**Comment by earlephilhower** [comment]
I know nothing about ModBus or that library, but you may want to grab a logic analyzer recording for working and non-working for the library authors.  Also, it looks like this is an Arduino.cc library so they may have special mbedOS-only options configured (we're bare metal, no OS).  If there are other OSS ModBus libs you may want to look at them.

**Reply by pomplesiegel** [reply]
Great idea on both fronts! Thank you. Anyone have experience with other likely compatible modbus libraries?

**Comment by pomplesiegel** [comment]
Hello! FYI, I switched to this library instead (below) and for me it is working much more reliably for the RP2040 on this platform. Thank you! 

https://github.com/CMB27/ModbusRTUSlave
